sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -197,16 +198,18 @@ class AioHttpStreamDownloadGenerator(AsyncIterator):
 
     :param pipeline: The pipeline object
     :param response: The client response object.
-    :param block_size: block size of data sent over connection.
-    :type block_size: int
+    :keyword bool decompress: If True which is default, will attempt to decode the body based
+        on the ‘content-encoding’ header.
     """
-    def __init__(self, pipeline: Pipeline, response: AsyncHttpResponse) -> None:
+    def __init__(self, pipeline: Pipeline, response: AsyncHttpResponse, **kwargs) -> None:
         self.pipeline = pipeline
         self.request = response.request
         self.response = response
         self.block_size = response.block_size
+        self._decompress = kwargs.pop("decompress", True)
+        if len(kwargs) > 0:
+            raise ValueError("Unknown parameters!")
```

Comment:
`TypeError` + include the first incorrect parameter name.

Reply:
Sorry, why TypeError? 

We expect the kwargs to be a single key dict by user passes in a multi-key dict, right?

Reply:
`kwargs` is not being passed in as a dict. Try the following:

```python
def tryit(a=None, b=None):
    print(a, b)

tryit(c=7)
```

Reply:
Updated.